### PR TITLE
chore(server): collapse project-ct and project-e2e into project-base

### PR DESCRIPTION
**Pull request by lmiller1990**
Currently you need to decide if you are working with a CT or E2E project when you open Cypress. 

This is problematic for the unification, since we won't know which type of project you'd like to work with until you select it from the unified GUI. In fact, the idea of "CT" or and "E2E" project doesn't make sense; what we really want is to create a project, then set the "mode" to running in "CT" or "E2E", depending on what the user decides they'd like to do. 

Also, you should be able to switch between the two without calling `openProject.create` again. This is useful for the scenario you are running some component tests, then you'd like to change to E2E, or vice versa.

Right now you need to decide which one you want when you `create` a project, which isn't really ideal: https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/open_project.js#L327-L332

Ideally, we should be able to do `openProject.create` which does all the common functionalities, like finding the spec files, etc. Later on, before we launch the runner, we can specify e2e/ct, and do the project specific behavior.

This PR reduces the gap between the two `Project` types, which will help in implementing some kind of API to allow switching between project types more seamlessly. Eventually, there should be little to no distinction between an E2E and CT project, except a few specific things like starting a `dev-server` for CT.

### CODE REVIEW

File: packages/server/tsconfig.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "./../ts/index.d.ts"
   ],
   "compilerOptions": {
-    "types": ["mocha", "node"]
+    "types": ["mocha", "node"],
+    "importHelpers": true
```
Comment (tgriesser) [inline packages/server/tsconfig.json:12]: Just need to add `tslib` as a dependency b/c of this change

File: packages/server/lib/project-base.ts [hunk 2]
```diff
@@ -155,7 +198,9 @@ export class ProjectBase<TServer extends ServerE2E | ServerCt> extends EE {
         return scaffold.plugins(path.dirname(cfg.pluginsFile), cfg)
       }
     })
-    .then(callbacks.onOpen)
+    .then((cfg) => {
+      return this.onOpen(cfg, options)
+    })
```
Comment (tgriesser) [inline packages/server/lib/project-base.ts:203]: Nice - would prefer methods over callback args when possible

**Review by tgriesser — DISMISSED** [review]

File: packages/server/tsconfig.json (hunk 1 above)
Comment (lmiller1990) [reply packages/server/tsconfig.json:12]: Will do.

**Comment by lmiller1990** [comment]
@tgriesser added `tslib`, can I get another quick 👀  and get this ✅  so I can :merge: (no emoji for merge unfortunately).

**Review by tgriesser — APPROVED** [review]

**Comment by tgriesser** [comment]
@lmiller1990 fine with me, CI still seems to be complaining about missing tslib though 🤔?

File: packages/server/package.json [hunk 3]
```diff
@@ -115,6 +115,7 @@
     "trash": "5.2.0",
     "tree-kill": "1.2.2",
     "ts-node": "8.5.4",
+    "tslib": "^2.3.0",
```
Comment (tgriesser) [inline packages/server/package.json:118]: Might need to re-yarn so the lockfile gets touched in order for this to get picked up, I'm guessing that the circle cache of yarn install is what's still causing the failures.

**Review by tgriesser — DISMISSED** [review]

**Comment by lmiller1990** [comment]
Ok @tgriesser, we are finally ✅ 

Going to need that approval again - thanks.

**Review by elevatebart — APPROVED** [review]

**Pull request merged**